### PR TITLE
for TRD, adding TRDArrayADC and its dependencies with some test stubs

### PR DESCRIPTION
**Pull request by bazinski**
Add TRDArrayADC with its dependencies.
Add some test stubs.
Includes ArraySignal, ArrayDictionary, FeeParam and their associated dependencies for compilation purposes.
A POC for documentation in Detectors/TRD/base/README.md  not extensive.

### CODE REVIEW

**Comment by sawenzel** [comment]
Any reason to close this?

**Comment by bazinski** [comment]
clean up all the commit messages, it looks messy, its compiling now finally

**Comment by sawenzel** [comment]
ok

**Comment by bazinski** [comment]
assuming this passes, can the commits be squashed by whom ever merges this ?

**Comment by sawenzel** [comment]
Yes, I can squash the commits. Waiting for Tom to take a look.

File: Detectors/TRD/base/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,12 @@ SET(SRCS
   src/TRDCommonParam.cxx
   src/TRDSimParam.cxx
   src/Digit.cxx
+  src/TRDArrayADC.cxx
```
[the remark this answers is not in the document]
Comment (bazinski) [reply Detectors/TRD/base/CMakeLists.txt]: the one is for backward compatibility in case something goes wrong to check they both perform the same.

**Comment by sawenzel** [comment]
@tdietel : Could you take a look here?
@bazinski : Can you fix the formatting problems?

**Comment by bazinski** [comment]
@sawenzel the only formatting checks I can see in build/o2checkcode/o2 are not for the code that I submitted in this PR. I can fix TRDCommonParam.h but not keen to touch the others.

**Comment by sawenzel** [comment]
I am not talking about o2checkcode. The clang-format test checks only lines of code changed in this PR and this is the one running on travis. You can find the log here:

https://travis-ci.org/AliceO2Group/AliceO2/jobs/485281664

This should be green before we can continue.

**Comment by tdietel** [comment]
This PR is mostly a port of classes from AliRoot, with an additional reimplementation of AliTRDarrayADC (as arrayADC, TRDarrayADC).

The comments at the top of (TRD)arrayADC.h/.cxx should indicate how they differ from the original implementation. They are also not documented so far, we must keep that on the to-do list for after the PR has been merged.

My understanding of the signal index is that it should not be necessary if we have an efficient implementation of the ADC array. As I can see it, it was only necessary in AliRoot to avoid looping over a 70MB array to find the <5% non-zero entries.If we only store the <5%, I do not see a need for the index.

File: Detectors/TRD/base/include/TRDBase/ArrayADC.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+namespace trd
+{
+
+///////////////////////////////////////////////
+//                                           //
+// Container class for ADC values            //
+//                                           //
+///////////////////////////////////////////////
+
+// ported from cblume original code in AliRoot, possibly some modifications
```
Comment (tdietel) [inline Detectors/TRD/base/include/TRDBase/ArrayADC.h]:
```suggestion
// reimplementation of original code by cblume in AliRoot
// uses STL vectors
```

Comment (tdietel) [reply Detectors/TRD/base/include/TRDBase/ArrayADC.h]: A bit more information about the memory organisation would be very helpful.

Comment (bazinski) [reply Detectors/TRD/base/include/TRDBase/ArrayADC.h]: Yes, the "possibly some modifications" was not the best formulation. It was intended to be edited later, once all the modifications were complete.

**Comment by sawenzel** [comment]
@tdietel : If you want comments to be addressed, maybe it is better to "request changes" before approving (otherwise it is a bit ambiguous how to continue).

**Comment by bazinski** [comment]
@sawenzel will combine the arrayadc and trdarrayadc, it was a test.
Need to rename the other without the trd prepended as well.
I thought only function names that started with a verb were to be lower, case? Will fix accordingly.
Will do the comments as well to remove abiguity in @tdietel approval

**Comment by bazinski** [comment]
@sawenzel I think that is ok now. 
method names are lower case except for where inheritence requires upper case, mostly Copy.
ArraySignal and ArrayDictionary and FeeParam were added. A start of a version of documentation was done for the purposes of discussion in Detectors/TRD/base/README.md
@tdietel I think this removes the ambiguity now?

**Comment by sawenzel** [comment]
Thanks for your patience. Merging (as a single clean commit).

**Pull request merged**